cmd/end2endtest/helpers.go

```diff
@@ -631,3 +631,53 @@ func cspGenProof(pid, voterKey []byte, csp *ethereum.SignKeys) (*apiclient.Censu
 
 	return &apiclient.CensusProof{Proof: caProofBytes}, nil
 }
+
+func (t *e2eElection) verifyVoteCount(nvotesExpected int) error {
+	startTime := time.Now()
+
+	// Wait for all the votes to be verified
+	for {
+		count, err := t.api.ElectionVoteCount(t.election.ElectionID)
+		if err != nil {
+			log.Warn(err)
+		}
+		if count == uint32(nvotesExpected) {
+			break
+		}
+
+		if err := t.api.WaitUntilNextBlock(); err != nil {
+			return fmt.Errorf("timeout waiting for next block")
+		}
+
+		log.Infof("verified %d/%d votes", count, nvotesExpected)
+		if time.Since(startTime) > t.config.timeout {
+			log.Fatalf("timeout waiting for votes to be registered")
+		}
+	}
+
+	log.Infof("%d votes registered successfully, took %s (%d votes/second)",
+		t.config.nvotes, time.Since(startTime), int(float64(nvotesExpected)/time.Since(startTime).Seconds()))
+
+	return nil
+}
+
+func (t *e2eElection) endElectionAndFetchResults() (*vapi.ElectionResults, error) {
+	// Set the account back to the organization account
+	t.api = t.api.Clone(t.config.accountPrivKeys[0])
```

Comment:
setting `t.api` you're also modifying the receiver :sweat_smile: 

try `api := t.api.Clone(t.config.accountPrivKeys[0])` and use `api`

Reply:
Yep, my bad   :sweat_smile: